### PR TITLE
Update auth providers 

### PR DESCRIPTION
**Pull request by zshainsky**
Freddy, this is the first pass at updating the auth providers. 

Still need to add: Google and CAS.

### CODE REVIEW

File: docs/guide/authentication/authentication_providers.md [hunk 1]
```diff
@@ -1,7 +1,111 @@
...
+1. Create an Application: Select the Applications menu > Add Application > Create Application
+2. Check the OpenID Connect radio button and select create
+3. Enter this value for the login redirect URL: `https://www.zepl.com/api/v2/authenticator.identity.zepl/callback/openidprovider`
+4. Navigate to the general settings for the new application that you created
+5. Select `Authorization Code` for Authorized grant types. The final application configuration should look as follows:
+
+  > Note: The "Initiate login URI" can be blank.
+
+  
+<img src="../../img/authentication/openid_connect/okta/okta_application_settings.png" class="image-box img-100"/></br>
```
Comment (f-ewald) [inline docs/guide/authentication/authentication_providers.md]: `</br>` should not be here. I am not sure about the image tag. `../../` looks wrong.

File: docs/guide/authentication/authentication_providers.md [hunk 2]
```diff
@@ -1,7 +1,111 @@
...
+4. Copy Application (client) ID below
+
+5. Copy Directory (tenant) ID below
+
+<img src="../../img/authentication/openid_connect/azure_ad/azure_ad_key.png" class="image-box img-100"/></br>
+
+
+Creating a Client Secret:
+1. Select Azure Active Directory > App Registrations > Select the app you just created
+
```
Comment (f-ewald) [inline docs/guide/authentication/authentication_providers.md]: No need for new line here.

File: docs/guide/authentication/authentication_providers.md [hunk 3]
```diff
@@ -1,7 +1,111 @@
...
+3. Copy the secret value below
+
+  > Note: Once you navigate away from this page you can no longer copy this value.
+
+<img src="../../img/authentication/openid_connect/azure_ad/azure_ad_secret.png" class="image-box img-100"/></br>
+
+#### In Zepl's Authentication Settings:
+1. Select Settings > Authentication > OpenID
+
```
Comment (f-ewald) [inline docs/guide/authentication/authentication_providers.md:142]: Same here.

File: docs/guide/authentication/authentication_providers.md [hunk 4]
```diff
@@ -1,7 +1,111 @@
...
+2. OpenID URL: `https://login.microsoftonline.com/{Your Directory (tenant) ID}/v2.0`. Replace "{Your Okta Account Name}" with the Directory (tenant) ID  from above
```
Comment (f-ewald) [inline docs/guide/authentication/authentication_providers.md]: `.` missing.

File: docs/guide/authentication/authentication_providers.md [hunk 5]
```diff
@@ -1,7 +1,111 @@
...
+[Username / Password](#user-pass)
 
-## Google
+[Google](#google)
 
-## OpenID
+[OpenID Connect](#openid-connect)
+  * [Okta](#openid-connect-okta)
+  * [Azure AD](#openid-connect-azure-ad)
+>The OpenID Connect protocol is supported by most SSO providers. If you don't see the name of your provider on this page, please check your provider's documentation.
```
Comment (f-ewald) [inline docs/guide/authentication/authentication_providers.md]: Space missing

File: docs/guide/authentication/authentication_providers.md [hunk 6]
```diff
@@ -1,7 +1,111 @@
...
 
-## CAS
+[CAS](#cas)
+<br>
+<br>
+
+# Username / Password <a name="user-pass"></a>
+
```
Comment (f-ewald) [inline docs/guide/authentication/authentication_providers.md:21]: Newline not required here.

File: docs/guide/authentication/authentication_providers.md [hunk 7]
```diff
@@ -1,7 +1,111 @@
...
+This is the default authentication method for Zepl and will be enabled upon sign up. This gives each user the ability to create a Zepl user name and password for authentication.
+
+<img src="../../img/authentication/zepl_username_password.png" class="image-box img-100"/></br>
```
Comment (f-ewald) [inline docs/guide/authentication/authentication_providers.md]: `</br>` should not be here

File: docs/guide/authentication/authentication_providers.md [hunk 8]
```diff
@@ -1,7 +1,111 @@
...
+
+  > Note: The "Initiate login URI" can be blank.
+
+  
+<img src="../../img/authentication/openid_connect/okta/okta_application_settings.png" class="image-box img-100"/></br>
+
+
+5. Scroll to the bottom to find the `Client ID` and `Client secret`. Please copy these or keep for reference in the next section
+
+<img src="../../img/authentication/openid_connect/okta/okta_client_credentials.png" class="image-box img-100"/></br>
```
Comment (f-ewald) [inline docs/guide/authentication/authentication_providers.md]: `</br>` should not be here

File: docs/guide/authentication/authentication_providers.md [hunk 9]
```diff
@@ -1,7 +1,111 @@
...
+2. OpenID URL: `https://{Your Okta Account Name}.okta.com`
+
+  >Note: This should NOT contain '-admin' or a trailing slash character (/) at the end of the url. Be sure to include the proper protocol, `http://` or `https://`.
+
+3. Client ID: Copy from the okta application you created in the previous section. Example: `4o199js5phG2Cv3204p6`
+4. Client Secret: Copy from the okta application you created in the previous section. Example: `0S-djdCveAac3aGz0CtNBOOwndoHL3I1MWbVv7el`
+5. Select Save & Activate
+6. Now, logout, log back in, and you should be redirected to Okta
+
+<img src="../../img/authentication/openid_connect/okta/okta_zepl_config_openid_connect.png" class="image-box img-100"/></br>
```
Comment (f-ewald) [inline docs/guide/authentication/authentication_providers.md]: `</br>` should not be here

File: docs/guide/authentication/authentication_providers.md [hunk 10]
```diff
@@ -1,7 +1,111 @@
...
+
+
+Creating a Client Secret:
+1. Select Azure Active Directory > App Registrations > Select the app you just created
+
+2. Select Certificates & secrets > + New client secret
+
+3. Copy the secret value below
+
+  > Note: Once you navigate away from this page you can no longer copy this value.
```
Comment (f-ewald) [inline docs/guide/authentication/authentication_providers.md]: Indentation not necessary here.

File: docs/guide/authentication/authentication_providers.md [hunk 11]
```diff
@@ -1,7 +1,111 @@
...
+
+3. Client ID: Copy the Application (client) ID from the previous section. Example: `d610128e-9dda-4575-9d2s-b5e75a1e457w`
+
+4. Client Secret: Copy the client secret from the previous section. Example: `VUq1jKjSsI9HaI1MtaSD3@oZFYfU/I_`
+
+5. Select Save & Activate
+
+6. Now, logout, log back in, and you should be redirected to the Azure authentication page
+
+<img src="../../img/authentication/openid_connect/azure_ad/azure_zepl_config_openid_connect.png" class="image-box img-100"/></br>
```
Comment (f-ewald) [inline docs/guide/authentication/authentication_providers.md]: `</br>` should not be here

File: docs/guide/authentication/authentication_providers.md [hunk 12]
```diff
@@ -1,7 +1,111 @@
...
+
+# CAS <a name="cas"></a>
+
+>Note: This documentation is coming soon to a Zepl docs page near you!
```
Comment (f-ewald) [inline docs/guide/authentication/authentication_providers.md]: Space is missing here.

**Review by f-ewald — COMMENTED** [review]
I reviewed the technical side and have a few comments. Please fix those and/or look them up. After that I will approve.

File: docs/guide/authentication/authentication_providers.md (hunk 1 above)
Comment (zshainsky) [reply docs/guide/authentication/authentication_providers.md]: What od you mean by "not sure about the image tag...looks wrong"? Should this be a fullpath? I copied the image tag from other docs pages. Let me know

Comment (zshainsky) [reply docs/guide/authentication/authentication_providers.md]: Removed `</br>`

File: docs/guide/authentication/authentication_providers.md [hunk 13]
```diff
@@ -1,7 +1,111 @@
...
+  > Note: Once you navigate away from this page you can no longer copy this value.
+
+<img src="../../img/authentication/openid_connect/azure_ad/azure_ad_secret.png" class="image-box img-100"/></br>
+
+#### In Zepl's Authentication Settings:
+1. Select Settings > Authentication > OpenID
+
+2. OpenID URL: `https://login.microsoftonline.com/{Your Directory (tenant) ID}/v2.0`. Replace "{Your Okta Account Name}" with the Directory (tenant) ID  from above
+
+  >Note: This should NOT contain a slash character (/) at the end of the url. Be sure to include the proper protocol, `http://` or `https://`.
```
Comment (zshainsky) [inline docs/guide/authentication/authentication_providers.md]: Added space "> Note:..."

**Comment by zshainsky** [comment]
Gents,

What is outstanding here so we can get this out on the public website?

**Review by louis-zepl — APPROVED** [review]

**Pull request merged**